plugins/intel-spi/fu-intel-spi-device.c

```diff
@@ -358,6 +362,11 @@ fu_intel_spi_device_setup(FuDevice *device, GError **error)
 		fu_device_add_child(device, child);
 	}
 
+	/* publish HSFS value for anyone else to use */
+	me_is_locked = !(self->hsfs & (1 << HSFS_FDV_OFF))     /* assume locked if not valid */
+	             || (self->hsfs & (1 << HSFS_FDOPSS_OFF)); /* use status bit if valid */
+	fu_context_set_me_lock_status(fu_device_get_context(device), me_is_locked);
```

Comment:
rather than using the context, could we use something like https://github.com/fwupd/fwupd/blob/main/libfwupdplugin/fu-device-metadata.h#L10 and then get access to the device metadata using `vfuncs->device_registered` in the other plugins? It seems to me this is more of the state of a specific device rather than a whole system attribute, although I can certainty argue it the other way around too.

Reply:
You can look at what plugin the device comes from.  There should only be one device from the intel-spi plugin, right?

Reply:
> There should only be one device from the intel-spi plugin, right?

I think it's one device but with one or more children, one of which actually corresponds to ME entry in IFD if it's present.

Reply:
Then my proposal was that you either add that attribute to the top level device and just look for the top level device from the plugin or you add it to a child and check for a device that has the metadata set.
If you end up needing a reference again in the plugin, you can always cache it locally in the plugin when you find it.

Reply:
Resolved last hardware issue and can now test it properly. The child device even shows up in `get-devices --show-all-devices` as "Intel Management Engine", so I'll mark it with metadata. Thanks.

Reply:
Another thing you can do is just use the locked device flag. Then no need to define a new metadata string.